### ISSUE TITLE
Bug: @typescript-eslint/utils - RulesTester.js makes import call invalid with eslint@8.29.0

### ISSUE THREAD

**Issue by meatnordrink** [typescript-eslint/typescript-eslint#6189]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
// @typescript-eslint/utils/dist/eslint-utils/rule-tester/RuleTester.js
ln 54 - const package_json_1 = require("eslint/package.json");
// eslint, as of v8.29 at least, is found in `@eslint`, no longer `eslint`. So this import is failing.
```

### Expected Result

The project to compile successfully. 

### Actual Result

Failed to compile, with the error message
```
Failed to compile.

[eslint] Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot find module 'eslint/package.json'
Require stack:
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/eslint-utils/rule-tester/RuleTester.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/eslint-utils/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/rules/adjacent-overload-signatures.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
Referenced from: /home/meatnordrink/gitRepos/landing-page-admin/.eslintrc.js
ERROR in [eslint] Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot find module 'eslint/package.json'
Require stack:
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/eslint-utils/rule-tester/RuleTester.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/eslint-utils/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/utils/dist/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/rules/adjacent-overload-signatures.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /home/meatnordrink/gitRepos/landing-page-admin/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
Referenced from: /home/meatnordrink/gitRepos/landing-page-admin/.eslintrc.js


```

### Additional Info

eslint, as of v8.29 at least, is found in `@eslint`, no longer `eslint`. So this import is failing.

```
// @typescript-eslint/utils/dist/eslint-utils/rule-tester/RuleTester.js
ln 54 - const package_json_1 = require("eslint/package.json");
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/utils`             | `5.46.0` |
| `ESLint`                               | `8.29.0` |
